service/header/sync.go

```diff
@@ -118,3 +93,56 @@ func (s *syncer) validator(ctx context.Context, p peer.ID, msg *pubsub.Message)
 	//  This introduces additional complexity for which we don't have time at the given moment.
 	return pubsub.ValidationIgnore
 }
+
+// getHead tries to get head locally and if not exists requests genesis.
+func (s *Syncer) getHead(ctx context.Context) (*ExtendedHeader, error) {
+	head, err := s.store.Head(ctx)
+	switch err {
+	case nil:
+		return head, err
+	case ErrNoHead:
+		// if there is no head - setup genesis.
+		genesis, err := s.exchange.RequestByHash(ctx, s.genesis)
+		if err != nil {
+			log.Errorw("requesting genesis", "err", err)
+			return nil, err
+		}
+
+		err = s.store.Append(ctx, genesis)
+		if err != nil {
+			log.Errorw("appending genesis to store", "err", err)
+			return nil, err
+		}
+	}
```

Comment:
this isn't really `genesis` right? We're just giving it a hash to start syncing *up to* rather than a genesis hash. might be confusing, but I don't care for devnet.

Reply:
related: https://github.com/celestiaorg/celestia-node/pull/220#discussion_r753100749